device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -429,4 +445,32 @@ public void registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeC
         //Codes_SRS_DEVICE_IO_34_020: [This function shall register the callback with the transport.]
         this.transport.registerConnectionStatusChangeCallback(statusChangeCallback, callbackContext);
     }
+
+    @Override
+    public void onTransportConnectionLost()
+    {
+        try
+        {
+            //only close the device io layer, not the transport layer. The transport layer is already closed when this method is called
+            this.close(false);
+        }
+        catch (IOException e)
+        {
+            log.warn("Encountered an exception while closing device IO layer due to transport disconnection event", e);
+        }
```

Comment:
This is bad pattern to catch exception and only log. it is absorbing the exception. What action can SDK user perform in this situation?

Reply:
This exception is only caught when opening/closing the transport layer. Since the transport layer isn't being opened/closed, this exception should never be caught

Reply:
In that case would it be safe to rethrow after logging?

Reply:
Upon reworking the DeviceIO layer, this is no longer an issue. No exceptions are being swallowed like this anymore.